src/bsoncxx/array/view.hpp

```diff
@@ -30,26 +30,91 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace array {
 
+///
+/// A read-only, non-owning view of a BSON document.
+///
 class BSONCXX_API view {
 
    public:
     class iterator;
     class const_iterator;
 
+    ///
+    /// @returns An const_iterator to the first element of the array.
+    ///
     const_iterator cbegin() const;
+
+    ///
+    /// @returns An const_iterator to the past-the-end element of the array.
+    ///
     const_iterator cend() const;
 
+    ///
+    /// @returns An iterator to the first element of the array.
+    ///
     iterator begin() const;
+
+    ///
+    /// @returns An iterator to the past-the-end element of the array.
+    ///
     iterator end() const;
 
+    ///
+    /// Indexes into this BSON array. If the index is out-of-bounds, a past-the-end iterator
+    /// will be returned. As BSON represents arrays as documents, the runtime of find() is
+    /// linear in the length of the array.
+    ///
+    /// @param i
+    ///   The index of the element.
+    ///
+    /// @return An iterator to the element if it exists, or the past-the-end iterator.
+    ///
     iterator find(std::uint32_t i) const;
+
+    ///
+    /// Indexes into this BSON array. If the index is out-of-bounds, the invalid element
+    /// will be returned. As BSON represents arrays as documents, the runtime of operator[] is
```

Comment:
Might want to specify the invalid element type so it creates a link to it in the generated output.